ee/clickhouse/queries/funnels/funnel_unordered.py

```diff
@@ -148,7 +148,7 @@ def _get_exclusion_condition(self):
             exclusion_time = f"exclusion_{exclusion_id}_latest_{exclusion.funnel_from_step}"
             condition = (
                 f"if( {exclusion_time} > {from_time} AND {exclusion_time} < "
-                f"if(isNull({to_time}), {from_time} + INTERVAL {self._filter.funnel_window_interval} DAY, {to_time}), 1, 0)"
+                f"if(isNull({to_time}), {from_time} + INTERVAL {self._filter.funnel_window_interval} {self._filter.funnel_window_interval_unit_ch()}, {to_time}), 1, 0)"
```

Comment:
Q: Are we interpolating user input in here? :monocle_face: 

Also, wouldn't it make more sense to use https://clickhouse.tech/docs/en/sql-reference/functions/type-conversion-functions/ `toIntervalX` functions? Would interpolate nicer and allow passing things in as parameters.

Reply:
Not really, it's this: https://github.com/PostHog/posthog/blob/master/posthog/models/filters/mixins/funnel.py#L100

This is part of generating the syntax, and Window functions don't support arbitrary computation (even the functions you mentioned) inside the `INTERVAL` yet.

Reply:
What you say makes a lot of sense 👀 .

Reply:
We could default to the smallest interval we have (`minutes`) in the API... and pass computation conversion requirements up?.

Or keep the API the same, and do the computation ourselves to only use MINUTES in the interval?

I don't see strong arguments for either, tbh, unless you have any?

Reply:
Either works for me I think - whatever helps us remove string interpolations :)

Reply:
Also, hmm, it's not a user string interpolation, but a restricted one. So, even keeping it as-is doesn't seem too different from the other options.

Curious, why do you want to remove this string interpolation?

Reply:
Well, every sql injection story starts with interpolation. In this case it's safe now, but future refactorings might break it.

When possible parameters > injection seems like a safe play, but willing to let this go if you think this approach is better.

Reply:
Makes sense. Will do it as a separate PR (there's a LOT of places using the same pattern right now, and we have to deprecate the old `funnel_window_days` once the frontend is updated)